v0/destinations/engage/transform.js

```diff
@@ -0,0 +1,214 @@
+
+const responseBuilder = (payload, endpoint, method, Config) => {
+  if (!payload) {
+    throw new CustomError("[ ENGAGE ]:: Parameters could not be found", 400);
+  }
+  const { publicKey, privateKey } = Config;
+  const response = defaultRequestConfig();
+  const basicAuth = Buffer.from(`${publicKey}:${privateKey}`).toString(
+    "base64"
+  );
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization: `Basic ${basicAuth}`
+  };
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.endpoint = endpoint;
+  response.method = method;
+  return response;
+};
+
+const identifyResponseBuilder = (message, Config) => {
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  const { traits } = message;
+  const uid = getUID(message);
+  // LIST FOR NOW IS IN CONFIG BUT TODO
+  const endpoint = `${ConfigCategories.IDENTIFY.endpoint.replace("uid", uid)}`;
+  const refinedPayload = refinePayload(
+    traits,
+    ConfigCategories.IDENTIFY.genericFields
+  );
+  set(payload, "meta", refinedPayload);
+  const listIds = getLists(message, Config);
+  if (listIds.length > 0) {
+    set(payload, "lists", listIds);
+  }
+  return responseBuilder(
+    payload,
+    endpoint,
+    ConfigCategories.IDENTIFY.method,
+    Config
+  );
+};
+
+const trackResponseBuilder = (message, Config) => {
+  if (!message.event) {
+    throw new CustomError("[ ENGAGE ]:: Event Name can not be empty.", 400);
+  }
+  const { properties } = message;
+  const payload = {};
+  const meta = refinePayload(properties, ConfigCategories.TRACK.genericFields);
+  const uid = getUID(message);
+  const endpoint = `${ConfigCategories.TRACK.endpoint.replace("uid", uid)}`;
+  payload.properties = meta;
+  payload.event = message.event;
+  payload.timestamp = message.originalTimestamp;
+  return responseBuilder(
+    payload,
+    endpoint,
+    ConfigCategories.TRACK.method,
+    Config
+  );
+};
+const pageResponseBuilder = (message, Config) => {
+  const uid = getUID(message);
+  const endpoint = `${ConfigCategories.PAGE.endpoint.replace("uid", uid)}`;
+
+  const { properties } = message;
+  const payload = {};
+  const meta = refinePayload(properties, ConfigCategories.PAGE.genericFields);
+  const pagePayload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.PAGE.name]
+  );
+  const mergedPayload = { ...meta, ...pagePayload };
+  payload.properties = mergedPayload;
+  payload.timestamp = message.originalTimestamp;
+  const name = getEvent(message);
+  payload.event = name;
+  return responseBuilder(
+    payload,
+    endpoint,
+    ConfigCategories.PAGE.method,
+    Config
+  );
+};
+
+const groupResponseBuilder = (message, Config) => {
+  const { groupId } = message;
+  if (!groupId) {
+    throw new CustomError("[ ENGAGE ]:: Group Id can not be empty.", 400);
+  }
+  const uid = getDestinationExternalID(message, "engageId");
```

Comment:
if not present, lets fallback to userId

Reply:
Okay

Reply:
So if are falling back to userId than we don't have any case without the Uid and so there is no scope to link a user to a list with the given email or phone number inside the traits.